plasmapy/analysis/nullpoint.py

```diff
@@ -1240,37 +1258,90 @@ def null_point_find(
     Returns
     -------
     array_like of `~plasmapy.analysis.nullpoint.NullPoint`
-        An array of NullPoint objects representing the nullpoints
-        of the given vector space.
+            An array of NullPoint objects representing the nullpoints
+            of the given vector space.
 
     Notes
-    -----
+    -------
     This method is described by :cite:t:`haynes:2007`.
+
     """
     # Constructing the vspace
     vspace = _vector_space(
         x_arr,
         y_arr,
         z_arr,
-        x_range,
-        y_range,
-        z_range,
+        None,
+        None,
+        None,
         u_arr,
         v_arr,
         w_arr,
+        None,
+        None,
+    )
+    return _vspace_iterator(vspace, MAX_ITERATIONS, err)
+
+
+def uniform_nullpoint_find(
+    x_range=[0, 1],
+    y_range=[0, 1],
+    z_range=[0, 1],
+    func=(lambda x, y, z: [x, y, z]),
```

Comment:
```suggestion
    x_range,
    y_range,
    z_range,
    func: Callable,
```
My inclination would be to not specify defaults for these parameters, and require users to specify them.

I'm also thinking it'd be helpful to specify `Callable` as a [type hint annotation](https://docs.python.org/3/library/typing.html#module-typing) for `func`.  This would require a `from typing import Callable` in the import section.